src/main.ts

```diff
@@ -10,15 +10,25 @@ import { loadExampleData, USERS } from "./model";
 import { UserRole, UserStatus } from "./grpc-gen/user";
 import { DELAYING_INTERCEPTOR } from "./interceptors/delaying-interceptor";
 import { logger } from "./logger";
-import { ENABLE_DUMMY_ADMIN, EXAMPLE_DATA_FILE, PORT, RANDOMNESS_SEED, WEB_PORT } from "./options";
+import {
+    ENABLE_DUMMY_ADMIN,
+    EXAMPLE_DATA_FILE,
+    WEB_ORIGIN,
+    PORT,
+    RANDOMNESS_SEED,
+    WEB_PORT,
+} from "./options";
 
 const ADDRESS = "0.0.0.0";
 const ENDPOINT = `${ADDRESS}:${PORT}`;
 
 proxy({
     target: `http://127.0.0.1:${PORT}`,
-    origin: "",
-    headers: "",
+    // Hack applied due to limiting types of @grpc-web/proxy. See cors library
+    // for more options.
+    /* eslint-disable  @typescript-eslint/no-explicit-any */
+    origin: WEB_ORIGIN as any,
```

Comment:
```suggestion
    origin: WEB_ORIGIN.source,
```

Can't you just do this?

Reply:
I do not think this is the same. The option is passed through to the `cors` package, which would interpret the string as a URL instead of a RegExp. See [Cors' Configuration Options](https://www.npmjs.com/package/cors#configuration-options) for the `origin` value.